presto-native-execution/presto_cpp/main/functions/remote/client/Remote.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "presto_cpp/main/functions/remote/client/Remote.h"
+
+#include "presto_cpp/main/functions/remote/client/RestRemoteClient.h"
+#include "velox/type/fbhive/HiveTypeSerializer.h"
+
+using namespace facebook::velox;
+namespace facebook::presto::functions {
+namespace {
+
+std::string serializeType(const TypePtr& type) {
+  // Use hive type serializer.
+  return type::fbhive::HiveTypeSerializer::serialize(type);
+}
+
+class PrestoRemoteFunction : public velox::exec::VectorFunction {
+ public:
+  PrestoRemoteFunction(
```

Comment:
@Joe-Abraham : It might be better to pass the RestRemoteClient as a parameter to the VectorFunction. That way the engine owns the RestClients and decide which to use when it gets a function invocation.

Since we have a single remote rest server in PrestoServer, having a single RestClient is sufficient for now. If we need to add security support etc for the Rest end point in can be done using a config in the server itself.